### ISSUE TITLE
Opening edex from rofi

### ISSUE THREAD

**Issue by Solace7** [GitSquared/edex-ui#636]
Technical information
**Using version:**
 - [x ] `master` (running from GitHub-published source code, currently `v3.0.0-pre`)
 - [ ] `latest` (latest release, currently `v2.2.2`)
 - [ ] `vX.X.X` (specify other version)

**Running on:**
 - [x ] Linux
 - [ ] Windows
 - [ ] macOS

**How comfortable you are with your system and/or IT in general:**
 - [ ] I'm kind of lost, honestly
 - [x ] I know what's up, I could help you run some commands or checks
 - [x ] My machine is fully under my control, tell me what you need
 - [ ] I attended [Defcon](https://defcon.org/) last year

---

## Problem

Something I started noticing back in v2.2.0, but I can't launch edex-ui from my rofi launcher anymore. I'm not sure if this is a rofi issue or an edex-ui issue. Launching from terminal works fine, it just means I have to keep the terminal window open.

Using rofis ability to open a terminal upon application launch (as I'm not sure if/where logs are for edex) I was able to capture this.

![2019-11-13-163902_1626x818_scrot](https://user-images.githubusercontent.com/9976088/68816904-d03dd280-0634-11ea-8733-39cf8ea280ac.png)

**Comment by GitSquared** [comment]
Could you give me the output of running `/usr/bin/zsh -ilc 'echo -n "_SHELL_ENV_DELIMITER_"; env; echo -n "_SHELL_ENV_DELIMITER_"; exit'`?

**Comment by Solace7** [comment]
_SHELL_ENV_DELIMITER_AUTOJUMP_ERROR_PATH=/home/solace/.local/share/autojump/errors.log
AUTOJUMP_SOURCED=1
BROWSER=/usr/bin/firefox
COLORTERM=truecolor
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
DESKTOP_SESSION=/usr/share/xsessions/awesome
DISPLAY=:0
EDITOR=vim
GTK_MODULES=canberra-gtk-module
HOME=/home/solace
LANG=en_US.UTF-8
LC_COLLATE=en_US.UTF-8
LC_CTYPE=en_US.UTF-8
LESS=-R
LOGNAME=solace
LSCOLORS=Gxfxcxdxbxegedabagacad
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
MAIL=/var/spool/mail/solace
MOZ_PLUGIN_PATH=/usr/lib/mozilla/plugins
OLDPWD=/home/solace
PAGER=less
PATH=/home/solace/.local/bin:/home/solace/bin/:/home/solace/.config/scripts/:/home/solace/.config/scripts/bin/:/home/solace/.gem/ruby/2.5.0/bin/:/home/solace/.local/bin:/home/solace/bin/:/home/solace/.config/scripts/:/home/solace/.config/scripts/bin/:/home/solace/.gem/ruby/2.5.0/bin/:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/var/lib/snapd/snap/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
PWD=/home/solace
SHELL=/bin/zsh
SHLVL=3
TERM=screen-256color
TMUX=/tmp/tmux-1000/default,12507,19
TMUX_PANE=%24
USER=solace
VTE_VERSION=5602
WINDOWID=48234499
XAUTHORITY=/home/solace/.Xauthority
XDG_CURRENT_DESKTOP=
XDG_DATA_DIRS=/usr/local/share:/usr/share:/var/lib/snapd/desktop
XDG_RUNTIME_DIR=/run/user/1000
XDG_SEAT=seat0
XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0
XDG_SESSION_CLASS=user
XDG_SESSION_DESKTOP=
XDG_SESSION_ID=1
XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0
XDG_SESSION_TYPE=x11
XDG_VTNR=1
ZSH=/home/solace/.oh-my-zsh
ZSH_TMUX_AUTOCONNECT=false
ZSH_TMUX_AUTOQUIT=true
ZSH_TMUX_AUTOSTART=true
ZSH_TMUX_AUTOSTARTED=true
ZSH_TMUX_TERM=screen-256color
_=/usr/bin/env
_ZSH_TMUX_FIXED_CONFIG=/home/solace/.oh-my-zsh/plugins/tmux/tmux.extra.conf
QT_QPA_PLATFORMTHEME=qt5ct
ICAROOT=/home/solace/Applications/ICAClient/
_SHELL_ENV_DELIMITER_

**Comment by GitSquared** [comment]
The error comes from [here](https://github.com/sindresorhus/shell-env/blob/a95fd441d3b7cc2c122899de72da30dae70936ec/index.js#L15), but I can't seem to reproduce it, even by emulating your env...

**Comment by Solace7** [comment]
Antergos. It's arch based.

**Comment by GitSquared** [comment]
Perhaps Rofi is doing some cleanup on the env before launching programs.

> `/usr/bin/zsh -ilc 'echo -n "_SHELL_ENV_DELIMITER_"; env; echo -n "_SHELL_ENV_DELIMITER_"; exit'`?

Could you try putting that into a `.sh` or something and launching it from Rofi this time?

If there's no difference then we'll have to look somewhere else.

**Comment by Solace7** [comment]
```
_SHELL_ENV_DELIMITER_AUTOJUMP_ERROR_PATH=/home/solace/.local/share/autojump/errors.log
AUTOJUMP_SOURCED=1
BROWSER=/usr/bin/firefox
COLORTERM=truecolor
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
DESKTOP_SESSION=/usr/share/xsessions/awesome
DISPLAY=:0
EDITOR=vim
GTK_MODULES=canberra-gtk-module
HOME=/home/solace
LANG=en_US.UTF-8
LC_COLLATE=en_US.UTF-8
LC_CTYPE=en_US.UTF-8
LESS=-R
LOGNAME=solace
LSCOLORS=Gxfxcxdxbxegedabagacad
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
MAIL=/var/spool/mail/solace
MOZ_PLUGIN_PATH=/usr/lib/mozilla/plugins
OLDPWD=/home/solace
PAGER=less
PATH=/home/solace/.local/bin:/home/solace/bin/:/home/solace/.config/scripts/:/home/solace/.config/scripts/bin/:/home/solace/.gem/ruby/2.5.0/bin/:/home/solace/.local/bin:/home/solace/bin/:/home/solace/.config/scripts/:/home/solace/.config/scripts/bin/:/home/solace/.gem/ruby/2.5.0/bin/:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/var/lib/snapd/snap/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
PWD=/home/solace
SHELL=/bin/zsh
SHLVL=3
TERM=screen-256color
TMUX=/tmp/tmux-1000/default,12507,40
TMUX_PANE=%48
USER=solace
VTE_VERSION=5602
WINDOWID=56623107
XAUTHORITY=/home/solace/.Xauthority
XDG_CURRENT_DESKTOP=
XDG_DATA_DIRS=/usr/local/share:/usr/share:/var/lib/snapd/desktop
XDG_RUNTIME_DIR=/run/user/1000
XDG_SEAT=seat0
XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0
XDG_SESSION_CLASS=user
XDG_SESSION_DESKTOP=
XDG_SESSION_ID=1
XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0
XDG_SESSION_TYPE=x11
XDG_VTNR=1
ZSH=/home/solace/.oh-my-zsh
ZSH_TMUX_AUTOCONNECT=false
ZSH_TMUX_AUTOQUIT=true
ZSH_TMUX_AUTOSTART=true
ZSH_TMUX_AUTOSTARTED=true
ZSH_TMUX_TERM=screen-256color
_=/usr/bin/env
_ZSH_TMUX_FIXED_CONFIG=/home/solace/.oh-my-zsh/plugins/tmux/tmux.extra.conf
QT_QPA_PLATFORMTHEME=qt5ct
ICAROOT=/home/solace/Applications/ICAClient/
_SHELL_ENV_DELIMITER_
```
Looks the same apart from some tmux information.

**Comment by Solace7** [comment]
I've installed dmenu which is similar to rofi and appear to have the same issue. So it isn't just rofi it seems. 
Another thing that I'm noticing is that running `edex-ui` from rofi/dmenu spawns a couple other processes `edex-ui --type=zygote`. When I run it from the command line itself though, other processes are also started.

I see in the terminal "success Settings loaded!" but after that nothing.

**Comment by GitSquared** [comment]
> Looks the same apart from some tmux information.

Indeed, everything works with this one too...

> running `edex-ui` from rofi/dmenu spawns a couple other processes `edex-ui --type=zygote`.

This is expected.

> I see in the terminal "success Settings loaded!" but after that nothing.

Not even an error?

**Comment by Solace7** [comment]
No error.
I did some extra testing. I firstly tried loading edex-ui from GNOMEs application launcher. This did the same thing as rofi and dmenu.
I then created a new user, tried and it works fine. So something specific to my users configuration is what's causing it and not necessarily anything wrong with edex-ui itself. The exact cause I have yet to find out.

Whatever the case may be it'll be something that a regular terminal doesnt interact with when trying to launch an application, but external "application launchers" do.
